### PR TITLE
MOBILE-1264_part2: Country & payment type selection

### PR DESCRIPTION
**Pull request by antons-zubkovs**

### CODE REVIEW

File: example/data/lib/api/api.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        try {
+          final parsedError = ApiErrorEntity.fromJson(parsedData);
+          throw ApiError(
+            code: parsedError.error.code,
+            name: parsedError.error.name,
+            description: parsedError.error.description,
+            statusCode: error.response?.statusCode,
+          );
+        } catch (_) {
+          // Ignore
```
Comment (PocketDaniel) [inline example/data/lib/api/api.dart:34]: What happens if app fails to parse `ApiErrorEntity`? Would user still be presented with error?

File: example/data/lib/payments/repository/network_payments_data_repository.dart [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import 'package:data/payments/api/payments_data_api_client.dart';
+import 'package:domain/payments/model/creditor.dart';
+import 'package:domain/payments/repository/payments_data_repository.dart';
+
+class NetworkPaymentsDataRepository extends PaymentsDataRepository {
```
Comment (PocketDaniel) [inline example/data/lib/payments/repository/network_payments_data_repository.dart:5]: This is a matter of preference, but in my previous experience we weren't specifying `Network` / `Local` (just assuming that repository that uses local storage is called `Local`).
So I would name this class as `PaymentsDataRepository` and class it extends `IPaymentsDataRepository ` where `I` indicates that this class is just an interface and doesn't contain any logic.
What are your thought on this?

**Review by PocketDaniel — CHANGES_REQUESTED** [review]

File: example/data/lib/api/api.dart (hunk 1 above)
Comment (antons-zubkovs) [reply example/data/lib/api/api.dart:34]: The generic error would thrown, check the end of the method bloc:

```dart
throw ApiError(
      statusCode: error.response?.statusCode,
      description: error.message,
      cause: error.error,
    );
```

Comment (PocketDaniel) [reply example/data/lib/api/api.dart:34]: Yes, makes sense. Just looks a bit odd

File: example/data/lib/payments/repository/network_payments_data_repository.dart (hunk 2 above)
Comment (antons-zubkovs) [reply example/data/lib/payments/repository/network_payments_data_repository.dart:5]: I believe that naming classes smth like `ClassImpl` or `ClassInterface` is not the best idea. Implementation should give a hint on how it's working, ex. `Local/Network/Persistent`. Not sure that it's worth naming interfaces with `I`, considering that in dart it's still an abstract class. What's important - to give a name, which tells client what data/functionality he gets.

**Review by PocketDaniel — APPROVED** [review]

**Pull request merged**